packages/typespec-autorest/test/openapi-output.test.ts

```diff
@@ -1031,4 +1038,90 @@ describe("identifiers decorator", () => {
       "cats/features/color",
     ]);
   });
+  it("`@identifiers` are assigned by model property", async () => {
+    const oapi = await openApiFor(
+      `
+      @armProviderNamespace
+      @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+      namespace Microsoft.Test;
+      
+      model Pet {
+        name: string;
+        id: int32;
+      }
+      model PetList {
+        @identifiers(#["name"])
+        value: Pet[]
+      }
+      model PetList2 {
+        @identifiers(#["id"])
+        value: Pet[]
+      }
+      `,
+    );
+    deepStrictEqual(oapi.definitions.PetList.properties.value["x-ms-identifiers"], ["name"]);
+    deepStrictEqual(oapi.definitions.PetList2.properties.value["x-ms-identifiers"], ["id"]);
+  });
+  it("`@identifiers` are assigned to `armProviderNamespace` properties even when nested in another namespace", async () => {
+    const oapi = await openApiFor(
+      `
+      @armProviderNamespace
+      @useDependency(Azure.ResourceManager.Versions.v1_0_Preview_1)
+      namespace Microsoft.Test;
+      
+      model PetResource is TrackedResource<PetResourceProperties> {
+        @key("petResourceName")
+        @segment("petServices")
+        name: string;
+      }
+
+      model PetResourceProperties {
+        @identifiers(#["key"])
+        pets?: PetTypes[];
+      }
+
+      model PetTypes {
+        key: string;
+        value: string;
+      }
+
+      @armResourceOperations(PetResource)
+      interface PetService {
+        update is ArmCustomPatchAsync<
+          PetResource,
+          Azure.ResourceManager.Foundations.ResourceUpdateModel<
+            PetResource,
+            PetResourceProperties
+          >
+        >;
+      }
+      `,
+    );
+    deepStrictEqual(oapi.definitions.PetResourceProperties.properties.pets["x-ms-identifiers"], [
+      "key",
+    ]);
+    deepStrictEqual(
+      oapi.definitions.PetResourceUpdateProperties.properties.pets["x-ms-identifiers"],
+      ["key"],
+    );
+  });
+  it("`@identifiers` are not supported outside @armProviderNamespace namespaces", async () => {
+    const diagnostics = await diagnoseOpenApiFor(
+      `
+      model Pet {
+        name: string;
+      }
+      model PetList {
+        @identifiers(#["name"])
+        value: Pet[];
+      }
+      `,
+    );
+
+    expectDiagnostics(ignoreDiagnostics(diagnostics, ["@typespec/http/no-service-found"]), [
```

Comment:
Why is this important?  Why do we want to prevent usage of this decorator outside ARM?

Reply:
we had a conversation some months ago where we decided this. I think it was because of something happening with the data-plane specs, but there was probably an underlying issue. Let me check that. I will be happy to remove this and make it not limited to ARM namespaces.

Reply:
ok confirmed that this is not needed, we can use it outside ARM, I did azure-rest-api-specs validation and everything looks good :)

Reply:
https://github.com/Azure/azure-rest-api-specs-pr/actions/runs/14538571923/job/40791853430?pr=22342